csxtools/utils.py

```diff
@@ -76,20 +77,28 @@ def get_fastccd_images(light_header, dark_headers=None,
 
         bgnd = np.array(dark)
 
-    data = _get_images(light_header, tag)
-    data = correct_images(data, bgnd, gain=gain)
+    data, n_images = _get_images(light_header, tag)
+    data = correct_images(data, bgnd, flat=flat, gain=gain)
     data = rotate90(data, 'cw')
-    return data
+    return data, n_images
 
 def _get_images(header, tag):
     t = ttime.time()
-    images = get_images(header, tag)[0]
+
+    all_event_images = get_images(header, tag)
+
+    images = []
+    n_images = []
+    for event_images in all_event_images:
+        images.append(np.asarray(event_images, np.uint16))
+        n_images.append(len(images[-1]))
```

Comment:
Could line 93 not be

```python
images = [np.asarray(event_images, np.uint16) for event_images in all_event_images]
```

I think that is cleaner. I still don't understand the use of `n_images`